lib/src/test/java/io/ably/lib/test/realtime/RealtimeAuthTest.java

```diff
@@ -1099,17 +1099,12 @@ public void onRawMessageRecv(ProtocolMessage message) {
             protocolListener.authCallback = params -> tokenDetails;
 
             final AblyRealtime ably = new AblyRealtime(protocolListener);
-            synchronized (protocolListener) {
-                ably.connect();
-                try {
-                    protocolListener.wait();
-                } catch (InterruptedException ie) {
-                    fail("auth_expired_token_expire_renew protocolListener.wait(): interrupted -" + ie.getMessage());
-                }
-            }
+            ably.connect();
 
             final Helpers.ConnectionWaiter connectionWaiter = new Helpers.ConnectionWaiter(ably.connection);
             boolean isConnected = connectionWaiter.waitFor(ConnectionState.connected, 1, 4000L);
+            //wait for the token to expire
+            wait(1000);
```

Comment:
A couple of observations about this time value:

1. It is floating here, disassociated from the token params' TTL defined earlier (maintainability)
2. It matches exactly the token params' TTL value which might be a risk of not waiting _quite enough_ at times, so could remain 'flakey' in nature 

So... I would, earlier on, create a `final` local variable containing the TTL value programmed into the token details. Then use that here, with a multiplier (perhaps x2 to be sure) - e.g. `tokenTtl * 2`.

Reply:
Resolving in @QuintinWillison's behalf as this is no longer the case